spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -371,6 +383,9 @@ public boolean tryLock(long time, TimeUnit unit) throws InterruptedException {
 				}
 				if (!acquired) {
 					localLock.unlock();
+					if (this.reLock <= 0) {
+						removeThreadLocal();
+					}
```

Comment:
The same is for this method too.
The typical pattern is:
```
if (lock.tryLock()) {
    try {
       .....
    } 
    finally {
       lock.unlock();
    }
}
```
So, if our `tryLock()` really fails with an exception we won't call `unlock` to clrear that `ThreadLocal`